gunicorn/instrument/statsd.py

```diff
@@ -98,29 +100,22 @@ def access(self, resp, req, environ, request_time):
     # statsD methods
     # you can use those directly if you want
     def gauge(self, name, value):
-        try:
-            if self.sock:
-                self.sock.send("{0}{1}:{2}|g".format(self.prefix, name, value))
-        except Exception:
-            pass
+        self._sock_send("{0}{1}:{2}|g".format(self.prefix, name, value))
 
     def increment(self, name, value, sampling_rate=1.0):
-        try:
-            if self.sock:
-                self.sock.send("{0}{1}:{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
-        except Exception:
-            pass
+        self._sock_send("{0}{1}:{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
 
     def decrement(self, name, value, sampling_rate=1.0):
-        try:
-            if self.sock:
-                self.sock.send("{0){1}:-{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
-        except Exception:
-            pass
+        self._sock_send("{0){1}:-{2}|c|@{3}".format(self.prefix, name, value, sampling_rate))
 
     def histogram(self, name, value):
+        self._sock_send("{0}{1}:{2}|ms".format(self.prefix, name, value))
+
+    def _sock_send(self, msg):
         try:
+            if isinstance(msg, six.text_type):
+                msg = msg.encode("ascii")
```

Comment:
Makes sense. Can you just confirm what those methods should return to clear up the confusion above?